doc/Reference/Markup/DependencyPropertyBuilder.md

```diff
@@ -0,0 +1,77 @@
+
+## Binding
+
+The `DependencyPropertyBuilder` provides a fluent API to help you create a binding. The `DependencyPropertyBuilder` is a generic class that takes the type of the property you are binding to as a generic parameter. This allows you to create a binding to a property of a specific type. The `DependencyPropertyBuilder` provides a fluent API to help you create a binding. The `DependencyPropertyBuilder` is a generic class that takes the type of the property you are binding to as a generic parameter. This allows you to create a binding to a property of a specific type.
+
+```cs
+new TextBlock()
+	.Text(x => x.Bind(() => vm.Message))
+```
+
+When creating a simplified binding such as the one above you may alternatively simply provide the binding expression like:
+
+```cs
+new TextBlock()
+	.Text(() => vm.Message);
+```
+
+It is also possible to provide more complex binding paths using either of the above Binding expressions like this:
+
+```cs
+new TextBlock()
+	.Text(() => vm.Client.Contact.FirstName)
+```
+
+#### Understanding the Binding Expression
+
+The Binding Expression is a lambda expression that provides the path to the property you are binding to. The lambda expression is parsed to determine the path to the property. The lambda expression can be a simple property access such as `() => vm.Message` or it can be a more complex expression such as `() => vm.Client.Contact.FirstName`. It is important to remember that this is equivalent to the following XAML:
+
+```xml
+<TextBlock Text="{Binding Message}" />
+<TextBlock Text="{Binding Client.Contact.FirstName}" />
+```
+
+While the Markup Extensions expose a delegate method for you to provide a path and strong typing for the property, it is easiest to think of the delegate as a string. If you think of the string value of `() => vm.Client.Contact.FirstName`, this will get evaluated by taking the substring following the first period leaving you with the path `Client.Contact.FirstName`. This is the path that will be used to create the binding.
+
+If you wish to perform tasks such as manipulate the value of the binding or format the text displayed see the [Converters](#converters) section below.
+
+### Converters
+
+The `DependencyPropertyBuilder`'s Bind method will return a strongly typed `BindingFactory`. This can help you as you tackle more advanced scenarios for Bindings such as by providing strongly typed converters that are picked up by the combination of the expected property type and the property type of the bound property. For instance you might have something like:
+
+```cs
+new TextBox()
+	.Text(() => vm.Query)
+	.Foreground(x => x.Bind(() => vm.Query)
+		.Convert(query => new SolidColorBrush(!string.IsNullOrEmpty(query) && query.Length > 5 ? Colors.Green : Colors.Red)));
```

Comment:
Should be within the Foreground(....) here ;)

Reply:
it is within the Foreground